google-cloud-spanner-hibernate-testing/src/test/java/org/hibernate/dialect/entities/JsonEntity.java

```diff
@@ -0,0 +1,127 @@
+package org.hibernate.dialect.entities;
+
+import com.google.cloud.spanner.hibernate.types.SpannerJsonType;
+import java.util.Objects;
+import java.util.UUID;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import org.hibernate.annotations.Type;
+import org.hibernate.annotations.TypeDef;
+import org.hibernate.annotations.TypeDefs;
+
+/**
+ * A Test entity with JSON entity fields.
+ */
+@TypeDefs({
```

Comment:
Why do we need this `@TypeDef` and `hibernate-types` extension doesn't?
See: https://github.com/vladmihalcea/hibernate-types/blob/master/hibernate-types-55/src/test/java/com/vladmihalcea/hibernate/type/json/MySQLGenericJsonTypeTest.java#L79.

Reply:
They seem to register types [programmatically](https://github.com/vladmihalcea/hibernate-types/blob/4ad5cec2f3539a1c5e95103849c215799b967d5f/hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/type/util/AbstractTest.java#L251) for [MySQL](https://github.com/vladmihalcea/hibernate-types/blob/1b0fb16c03c4685488f6b9c67f0843c07117fdcb/hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/type/json/MySQLParametrizedJsonTypeTest.java#L36) instead of with annotations.

There are [other types](https://github.com/vladmihalcea/hibernate-types/blob/6020ac94c530ac75b1ab70b7e2b0b28a5ee7f99f/hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/type/json/PostgreSQLJsonMapTest.java#L63) where the declarative TypeDef approach is used.

Reply:
Thanks for the clarification. I also now found that `@TypeDef` is required according to [this article](https://vladmihalcea.com/how-to-map-json-objects-using-generic-hibernate-types/).